This adds http routes for drone control

thewrongjames opened the pull request:
These commits allow arrays of json objects of the form described in issue #10 to be sent to the server hosted by the node app and control connected drones. It gives access to all the high level movement controls exposed by the node [ar-drone module](https://github.com/felixge/node-ar-drone), and also a light flashing command. This will close #10, and is an important step towards basic user-frendly controls.

ennuuos commented on drones.js:
```diff
@@ -88,6 +89,7 @@ class Drones {
     drone.id = id
     drone.ip = ip
     drone.connected = false
+    drone.flash = function () { this.animateLeds('blinkOrange', 5, 1) }
```
This function is not mentioned in the commit and is unrelated to the change. Please avoid this in future.

ennuuos commented on drones.js:
```diff
@@ -107,7 +109,7 @@ class Drones {
     drone.resume()
     drone.connected = true
-    drone.animateLeds('blinkOrange', 5, 1) // This animation lets us know the drone has connected
+    drone.flash() // This animation lets us know the drone has connected
```
This function call is not mentioned in the commit and is unrelated to the change. Please avoid this in future.

ennuuos commented on index.js:
```diff
@@ -1,7 +1,7 @@
-let server = require('./server')
+const server = require('./server')
```
This change is not mentioned in the commit and is unrelated to the change. Please avoid this in future.

ennuuos commented on index.js:
```diff
@@ -1,7 +1,7 @@
 const Drones = require('./drones')
 const config = require('./config.json')
-let drones = new Drones({
+const drones = new Drones({
```
This function is not mentioned in the commit and is unrelated to the change. Please avoid this in future.

ennuuos commented on drones.js:
```diff
@@ -198,23 +198,49 @@ class Drones {
     })
     server.post('/drones/:id', (req, res) => {
       let id = req.params['id']
+      console.log(this.add)
```
Please explain this line of code.

ennuuos replied:
Unnecessary logs should be avoided at all costs because they are a huge pain to track down.
If this is a useful output, add a check to the log variable in config.json

ennuuos commented on server.js:
```diff
@@ -14,4 +14,6 @@ server.listen(config.network.port, () => {
   console.log(`[${requestAddress}] ${req.method} ${req.url}`)
 })
 
+server.use(express.json())
```
While you're already making changes, please add a comment here about what this line does (in case someone doesn't know / forgets what this is)

thewrongjames replied:
```js
server.use(express.json()) // Uses json
```
?

thewrongjames replied:
That was a joke, you have a point.

ennuuos commented:
When the two changes have been made feel free to merge.

thewrongjames commented:
I have now addressed all the issues accept those involving the `.flash` function, which while presuasive (I should have been more explicit in my commit messages, and was likely wrong to assume that calling it 'command routing' covered the whole command route from client to drone), I don't believe are strong enough of arguments to warrant removing those changes. Is it acceptable to merge now @ennuuos?

ennuuos commented:
Accepted

ennuuos commented:
Please squash and merge, to remove the clutter. In the squashed commit please make note of the newly created flash abstraction.

thewrongjames commented:
Will do. I will also close dev after this merge, as pull requests have made it obvious that only merging from a single branch into master is not ideal. It seems we should have individual feature branches that make pull requests to master.

thewrongjames commented:
I have only just realised that the degree passed in to the user is not validated as being between 0 and 1 inclusive, nor is it even validated as being a number. I shall add a commit that uses the default degree if the passed in degree is invalid, and then I shall squash and merge.
